echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+                    ).opts(
+                        line_dash="dashed", line_width=2, color="red", alpha=0.75, show_legend=True
+                    )
+                )
+            # ---- Layer the data
+            return hv.Overlay(plot_layers).opts(height=800, width=800, xlabel=xlab, ylabel=ylab)
+
+        # Return the stacked Column object
+        return pn.Column(var_dropdown, update_plot)
+
+    def mesh_regions(self) -> hv.core.overlay.Overlay:
+
+        # Get the full dataset
+        DATAINPUT = self.data.analysis["kriging"]["transect_mesh_regions_df"].copy()
+        # ---- Get color code for each region
+        colormap = get_discrete_color_palette(len(DATAINPUT.mesh_region.unique()))
+        # ---- Apply the colormap
+        DATAINPUT["region_color"] = DATAINPUT["mesh_region"].apply(lambda x: colormap[x - 1])
+
+        # Format tooltip
+        TOOLTIPS = """
+            <div style="position: relative;">
+                <style>
+                    div.bk-tooltip-content > div > div:not(:first-child) {{
+                        display:none !important;
+                    }}
+                </style>
+                <div>
+                    <b>Transect: </b> <span>@transect_num</span> <br>
+                </div>
+            </div>
+            """
+        # ---- Create hovertool
+        hover_tool = HoverTool(tooltips=TOOLTIPS, mode="mouse")
+
+        # Create hovertool data
+        HOVERPATH = gv.Points(
+            DATAINPUT, kdims=["longitude", "latitude"], vdims=["transect_num"]
+        ).opts(alpha=0.0, tools=[hover_tool])
+
+        # Initialize plot layers
+        plot_layers = []
+
+        # Iterate and add colored regions
+        for region in DATAINPUT["mesh_region"].unique():
+            # ---- Format region label
+            region_label = f"Region {region}"
+            # ---- Subset the data
+            subset = DATAINPUT.loc[DATAINPUT["mesh_region"] == region]
+            # ---- Pull the color label
+            id_color = subset["region_color"].iloc[0]
+
+            #
+            plot_layers.append(
+                gv.Points(
+                    subset,
+                    kdims=["longitude", "latitude"],
+                    vdims=["transect_num"],
+                    label=region_label,
+                ).opts(
+                    color=id_color,
+                    show_legend=True,
+                )
+            )
+
+        # Stack the plotting layers for visualization
+        # ---- Stack
+        plot_stacked = (self.tile * HOVERPATH * gv.Overlay(plot_layers)).opts(
+            width=800, height=800, xlabel="Longitude (\u00B0E)", ylabel="Latitude (\u00B0N)"
+        )
+        # ---- Return
+        return plot_stacked
+
+    def nasc_map(self) -> hv.core.overlay.Overlay:
+
+        # Get the full dataset
+        DATAINPUT = self.data.analysis["transect"]["acoustics"]["adult_transect_df"].copy()
```

Comment:
I wonder if it is useful to make `"adult_transect_df"` the default but allow plotting other df names -- I forgot if there are other dfs in `self.data.analysis["transect"]["acoustics"]`.

Reply:
There are a few other dataframes that get added to the `"acoustics"` dictionary (i.e. `sigma_bs` averaged over hauls and strata), but those are more tabular and only provide the mean values (I think it would only be useful to have a dataviz for these if there were standard errors or some other uncertainty interval associated with the values).